File: frontend/src/types/account.ts
/**
 * Account types for MyBudget.
 */

import type { ConnectionHealthStatus } from './bankConnection'

export type AccountType = 'CHECKING' | 'SAVINGS'

export type SyncStatus = 'SUCCESS' | 'FAILED' | 'PENDING' | 'NEVER_SYNCED'

export interface Account {
  id: string
  user_id: string
  name: string
  account_type: AccountType
  balance: string
  initial_balance: string
  sync_status: SyncStatus
  last_sync_at: string | null
  sync_error: string | null
  created_at: string
  updated_at: string
  // Bank connection fields (present for linked accounts)
  bank_connection_id?: string | null
  bank_connection_name?: string | null
  bank_connection_health?: ConnectionHealthStatus | null
}

export interface AccountCreate {
  name: string
  account_type: AccountType
  initial_balance: string
}

export interface AccountUpdate {
  name?: string
}

export interface AccountListResponse {
  accounts: Account[]
  total: number
}

File: backend/src/mybudget/schemas/account.py
"""
Account Pydantic schemas for API request/response validation.
"""
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from mybudget.models.account import AccountType, SyncStatus
from mybudget.schemas.bank_connection import ConnectionHealthStatus


class AccountCreate(BaseModel):
    """Schema for creating a new account."""

    name: str = Field(..., min_length=1, max_length=100, description="Account name")
    account_type: AccountType = Field(..., description="Account type (CHECKING or SAVINGS)")
    initial_balance: Decimal = Field(..., description="Initial account balance")


class AccountUpdate(BaseModel):
    """Schema for updating an account."""

    name: str | None = Field(None, min_length=1, max_length=100, description="Account name")


class AccountResponse(BaseModel):
    """Schema for account response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    name: str
    account_type: AccountType
    balance: Decimal
    initial_balance: Decimal
    created_at: datetime
    updated_at: datetime
    sync_status: SyncStatus
    last_sync_at: datetime | None = None
    sync_error: str | None = None
    # Bank connection fields (for linked accounts)
    bank_connection_id: UUID | None = None
    bank_connection_name: str | None = None
    bank_connection_health: ConnectionHealthStatus | None = None


class AccountListResponse(BaseModel):
    """Schema for list of accounts response."""

    accounts: list[AccountResponse]
    total: int
